sfmutils/exporter.py

```diff
@@ -302,18 +330,33 @@ def id_field(self):
         pass
 
     def __iter__(self):
-        # yield the header row
-        yield self._header_row()
-        for post in self.warc_iter_cls(self.warc_paths,
-                                       self.seed_uids).iter(dedupe=self.dedupe,
-                                                            item_date_start=self.item_date_start,
-                                                            item_date_end=self.item_date_end,
-                                                            limit_item_types=self.limit_item_types):
+        warcs = self.warc_iter_cls(self.warc_paths,
+                                   self.seed_uids).iter(dedupe=self.dedupe,
+                                                        item_date_start=self.item_date_start,
+                                                        item_date_end=self.item_date_end,
+                                                        limit_item_types=self.limit_item_types)
+        iterator_warc = iter(warcs)
+        split_size = self.segment_row_size - 1 if self.segment_row_size != -1 else None
+        # make the iterator warc to chunks based on the row size
+        for post in iterator_warc:
             try:
-                yield self._row(post.item)
+                # define the chunk
+                def chunk():
+                    # yield the header row
+                    yield self._header_row()
+                    # get the first row in for loop
+                    yield self._row(post.item)
+                    # get the left more rows
+                    for more in islice(iterator_warc, split_size):
+                        yield self._row(more.item)
+
+                yield chunk()
             except KeyError, e:
                 log.warn("Invalid key %s in %s", e.message, json.dumps(post.item, indent=4))
 
+    def set_segment_size(self, row_size):
```

Comment:
Set this in __init__ rather than a separate setter method.  (Yes, I know it will require changing each of the exporter subclasses.)